primitives/messages/src/lib.rs

```diff
@@ -369,6 +370,8 @@ impl<RelayerId> InboundLaneData<RelayerId> {
 	{
 		relayers_entries
 			.checked_mul(UnrewardedRelayer::<RelayerId>::max_encoded_len())?
+			.checked_add(Compact::<u32>(relayers_entries as u32).encoded_size())?
+			.checked_add(LaneState::max_encoded_len())?
```

Comment:
> Also maybe we could have a macro that iterates through all the fields of a struct and generates this encoded_size_hint() method automatically.

There's similar trait + auto-derive already - `MaxEncodedLen`. However, it doesn't work well when we need to compute actual size. The alternate method would be to simply call the `.encoded_size()` on already available variable. But it does memory allocations + we don't always have this "variable" when we need to compute size.

This particular error is not critical actually. Before `MaxEncodedLen` was introduced, we already have our own bridge-specific `trait Size`, which was used in weight formulae. So it doesn't need to be super-accurate - it just needs to give a good estimation. Like - there's a test for this `encoded_size_hint`, which allow `10%` deviations. And it was not ever failing :) I've added this change just because I've accidentally had a typo and was using different types in this test: actual struct was using `u64` in struct and `u8` in computations. It isn't actual anymore (because `RewardAtSource` is now fixed to `u64`), but still decided to fix that :)